### ISSUE TITLE
twelf build broken

### ISSUE THREAD

**Issue by gebner** [NixOS/nixpkgs#8761]
Trying to run twelf in current master fails with:
```
$ nix-shell -p twelf --command twelf-server
/nix/store/ggmpcig6pjk0462ax2p0s3hcw35aayyx-smlnj-110.78/bin/sml: Fatal error -- unable to open heap image "/nix/store/8cgz8lcx38sly2na7695c4lp0prvn611-twelf-1.7.1//bin/.heap/twelf-server"
```

Apparently this is due to a [compiler bug that is ignored during the twelf build](http://hydra.nixos.org/build/22498097/log#line-6984).

Looking through the hydra build logs, this seems to be a regression with the update of smlnj from 110.76 to 110.78.

**Comment by jwiegley** [comment]
@vcunat Yes, I think so.

**Comment by pSub** [comment]
Fixed in 54dab782bea50e03d31d241c72eda16b799b8deb. Has anyone created an upstream bug?

**Comment by gebner** [comment]
Thanks for fixing this!

This is the SMLNJ bug: http://smlnj-gforge.cs.uchicago.edu/tracker/index.php?func=detail&aid=140&group_id=33&atid=215
